src/nvim/terminal.c

```diff
@@ -526,7 +526,11 @@ void terminal_paste(long count, char_u **y_array, size_t y_size)
     for (size_t j = 0; j < y_size; j++) {
       if (j) {
         // terminate the previous line
-        terminal_send(curbuf->terminal, "\n", 1);
+# ifdef UNIX
```

Comment:
I'm not sure what are the different possible platform are. But would it be possible to have a non-unix platform that doesn't use `\r\n`?

If this is the case, it would mean they would fall under the `#else` and receive the wrong characters